### DISCUSSION TITLE
Is it OK to call `reset` from a `onReset` form event-handler?

### DISCUSSION THREAD

**Discussion by Svish** [react-hook-form/react-hook-form#8520] [Q&A]
The documentation of the [`reset` function](https://react-hook-form.com/api/useform/reset), under "Rules", it mentions that "It's recommended to `reset` inside `useEffect` after submission", and to "Avoid calling `reset` before `useForm`'s `useEffect` is invoked.

I'd like to make use of the built-in reset function in forms, which does reset the contents of the input fields, but (of course) not the `react-hook-form` state. Is it safe and OK to call `reset` in an `onReset` event-handler as well, or only inside `useEffect`?

```ts
const { handleSubmit, reset } = useForm();

return (
  <form
    onSubmit={handleSubmit(someSubmitHandler)}
    onReset={() => { reset() }
  >
   
    <button type="reset">Reset</button>
    <button type="submit">Submit</button>
  </form>
)
```

**Comment by bluebill1049** [comment]
`useEffect` would be the safer option, because you do have micro tasks running inside the handlesSubmit function.
